File: personalizatio-sdk/src/main/java/com/personalizatio/sdk/Personaclick.java
package com.personalizatio.sdk;

import android.content.Context;

/**
 * Created by Sergey Odintsov
 *
 * @author nixx.dj@gmail.com
 */
final public class Personaclick extends SDK {

	public static final String TAG = "PERSONACLICK";
	private static final String PREFERENCES_KEY = "personaclick.sdk";
	public static final String NOTIFICATION_URL = "PERSONACLICK_NOTIFICATION_URL";

	/**
	 * @param context application context
	 * @param shop_id Shop key
	 */
	protected Personaclick(Context context, String shop_id) {
		super(context, shop_id, Api.PC.class);
	}

	@Override
	protected String getPreferencesKey() {
		return PREFERENCES_KEY;
	}

	@Override
	public String getTag() {
		return TAG;
	}

	/**
	 * Initialize api
	 * @param shop_id Shop key
	 */
	public static void initialize(Context context, String shop_id) {
		instance = new Personaclick(context, shop_id);
	}
}

File: personalization-sdk/src/personaclick/kotlin/com/personaclick/sdk/Personaclick.kt
package com.personaclick.sdk

import android.content.Context
import com.personalization.BuildConfig
import com.personalization.SDK

class Personaclick private constructor() : SDK() {

    companion object {
        private val DOMAIN_API: String = when {
            BuildConfig.DEBUG -> "192.168.1.8:8080"
            else -> "api.personaclick.com"
        }

        fun getInstance(): SDK = instance

        /**
         * Returns the SDK instance registered for the given shopId, or null if not found.
         */
        fun getInstance(shopId: String): SDK? = SDK.getInstance(shopId)

        /**
         * Initialize api
         * @param context application context
         * @param shopId Shop key
         */
        fun initialize(
            context: Context,
            shopId: String,
            apiHost: String? = null
        ) {
            val sdk = getInstance()

            initSdk(
                sdk = sdk,
                context = context,
                shopId = shopId,
                apiHost = apiHost
            )
        }

        /**
         * Creates and initializes a new SDK instance for the given shopId.
         * Use this for multi-instance scenarios where each shop needs its own SDK.
         *
         * @param context application context
         * @param shopId Shop key
         * @param apiHost optional API host override
         * @return initialized SDK instance
         */
        fun initializeInstance(
            context: Context,
            shopId: String,
            apiHost: String? = null
        ): SDK {
            val sdk = SDK()

            initSdk(
                sdk = sdk,
                context = context,
                shopId = shopId,
                apiHost = apiHost
            )

            return sdk
        }

        private fun initSdk(
            sdk: SDK,
            context: Context,
            shopId: String,
            apiHost: String?
        ) = sdk.initialize(
            context = context,
            shopId = shopId,
            apiDomain = apiHost ?: DOMAIN_API
        )
    }
}
